### PR TITLE
fix: only redirect for GET/HEAD requests

### LINKED ISSUE #12701: AWS SDK STS requests get redirected to console when running in browser
## Expected Behavior

It shouldn't be redirected.

## Current Behavior

It works when I set MINIO_BROWSER=off, and gets redirected to console when MINIO_BROWSER=on

## Steps to Reproduce (for bugs)

The following is what aws-sdk does in simple code
```javascript
const body = new URLSearchParams()
body.append("Action", "AssumeRoleWithWebIdentity")
body.append("WebIdentityToken", "xxx")
body.append("Version", "2011-06-15")
body.append("DurationSeconds", "86000")

fetch(`https://s3.example.com`, {
      method: 'POST',
      body
}).then()
```
The problem is, in either environment (MINIO_BROWSER=off and MINIO_BROWSER=off). The following code works:
```javascript
const body = new URLSearchParams()
body.append("Action", "AssumeRoleWithWebIdentity")
body.append("WebIdentityToken", "xxx")
body.append("Version", "2011-06-15")
body.append("DurationSeconds", "86000")

fetch(`https://s3.example.com/?${body.toString()}`, {
      method: 'POST'
}).then()
```

## Context

I want to use aws-sdk wrapper for this api to make my code consistency.

## Your Environment

* Version used (`minio --version`): https://dl.min.io/server/minio/release/linux-amd64/minio_20210712024453.0.0_amd64.deb
* AWS SDK version: aws-sdk@22.944.0 (javascript)

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: only redirect for GET/HEAD requests

## Motivation and Context
fixes #12701

## How to test this PR?
As per #12701 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Review by krisis — APPROVED** [review]

File: cmd/generic-handlers.go [hunk 1]
```diff
@@ -154,15 +154,15 @@ func setRedirectHandler(h http.Handler) http.Handler {
 
 func guessIsBrowserReq(r *http.Request) bool {
 	aType := getRequestAuthType(r)
-	ok := strings.Contains(r.Header.Get("User-Agent"), "Mozilla") && globalBrowserEnabled &&
-		(aType == authTypeJWT || aType == authTypeAnonymous)
-	return ok
+	return strings.Contains(r.Header.Get("User-Agent"), "Mozilla") &&
+		globalBrowserEnabled && aType == authTypeAnonymous
```
Comment (vadmeste) [inline cmd/generic-handlers.go:158]: Maybe you need to remove this as well ?  aType == authTypeAnonymous

It is weird that guessIsBrowserReq doesn't return true when this is authenticated request from the browser

**Review by vadmeste — COMMENTED** [review]
One comment

File: cmd/generic-handlers.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/generic-handlers.go:158]: There is no authentication coming to MinIO @vadmeste on browser anymore

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**